### PR TITLE
backgroud support for issue filing

### PR DESCRIPTION
**Pull request by haonliu**
#### Description of changes

backgroud support for issue filing

#### Pull request checklist

- [x] Added relevant unit test for your changes. (`npm run test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] Ran precheckin (`npm run precheckin`)

### CODE REVIEW

**Comment by smoralesd** [comment]
Did you not do any of the steps on the checklist? can you update it?

**Comment by haonliu** [comment]
> Did you not do any of the steps on the checklist? can you update it?

Done!

File: src/background/actions/action-creator.ts [hunk 1]
```diff
@@ -115,6 +121,10 @@ export class ActionCreator {
         this.registerTypeToPayloadCallback(Messages.SettingsPanel.OpenPanel, this.onOpenSettingsPanel);
         this.registerTypeToPayloadCallback(Messages.SettingsPanel.ClosePanel, this.onCloseSettingsPanel);
 
+        this.registerTypeToPayloadCallback(Messages.IssueFiling.OpenDialog, this.onOpenIssueFilingDialog);
```
Comment (smoralesd) [inline src/background/actions/action-creator.ts]: can you move this to it's own action creator

File: src/background/chrome-feature-controller.ts [hunk 2]
```diff
@@ -13,4 +13,8 @@ export class ChromeFeatureController {
     public openCommandConfigureTab(): void {
         this._browserAdapter.createTab(ChromeFeatureController.configureCommandTabUrl);
     }
+
+    public openIssueFilingWindow(url: string): void {
```
Comment (smoralesd) [inline src/background/chrome-feature-controller.ts]: why not use browser adapter directly? I don't think this is necessary.

File: src/background/actions/issue-filing-action-creator.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    constructor(
+        private issueFilingActions: IssueFilingActions,
+        private browserAdapter: BrowserAdapter,
+        private telemetryEventHandler: TelemetryEventHandler,
+        private registerTypeToPayloadCallback: RegisterTypeToPayloadCallback,
+        private detailsViewController: DetailsViewController,
+    ) {}
+
+    public registerCallbacks(): void {
+        this.registerTypeToPayloadCallback(Messages.IssueFiling.OpenDialog, this.onOpenIssueFilingDialog);
```
Comment (devaradhanm) [inline src/background/actions/issue-filing-action-creator.ts]: call it openSettingsDialog & closeSettingsDialog.
also rename onOpenIssueFilingDialog to onOpenIssueFilingSettingsDialog

**Comment by haonliu** [comment]
Close this PR for now.

**Pull request closed without merging**